src/core/MOM_open_boundary.F90

```diff
@@ -6145,6 +6156,102 @@ subroutine rotate_OBC_segment_data(segment_in, segment, turns)
+         call rotate_array(segment_in%rx_norm_rad, turns, segment%ry_norm_rad)
+    if (allocated(segment_in%ry_norm_rad)) &
+         call rotate_array(segment_in%ry_norm_rad, turns, segment%rx_norm_rad)
+    if (allocated(segment_in%rx_norm_obl)) &
+         call rotate_array(segment_in%rx_norm_obl, turns, segment%ry_norm_obl)
+    if (allocated(segment_in%ry_norm_obl)) &
+         call rotate_array(segment_in%ry_norm_obl, turns, segment%rx_norm_obl)
+  else
+    if (allocated(segment_in%rx_norm_rad)) &
+         call rotate_array(segment_in%rx_norm_rad, turns, segment%rx_norm_rad)
+    if (allocated(segment_in%ry_norm_rad)) &
+         call rotate_array(segment_in%ry_norm_rad, turns, segment%ry_norm_rad)
+    if (allocated(segment_in%rx_norm_obl)) &
+         call rotate_array(segment_in%rx_norm_obl, turns, segment%rx_norm_obl)
+    if (allocated(segment_in%ry_norm_obl)) &
+         call rotate_array(segment_in%ry_norm_obl, turns, segment%ry_norm_obl)
+  endif
+  if (allocated(segment_in%cff_normal)) &
+       call rotate_array(segment_in%cff_normal, turns, segment%cff_normal)
+  if (allocated(segment_in%nudged_normal_vel)) &
+       call rotate_array(segment_in%nudged_normal_vel, turns, segment%nudged_normal_vel)
+  if (allocated(segment_in%nudged_tangential_vel)) &
+       call rotate_array(segment_in%nudged_tangential_vel, turns, segment%nudged_tangential_vel)
+  if (allocated(segment_in%nudged_tangential_grad)) &
+       call rotate_array(segment_in%nudged_tangential_grad, turns, segment%nudged_tangential_grad)
+  if (associated(segment_in%tr_Reg)) then
+    do n = 1, segment_in%tr_Reg%ntseg
+      call rotate_array(segment_in%tr_Reg%tr(n)%t, turns, segment%tr_Reg%tr(n)%t)
+      call rotate_array(segment_in%tr_Reg%tr(n)%tres, turns, segment%tr_Reg%tr(n)%tres)
+      ! Testing this to see if it works for contant tres values. Probably wrong otherwise.
+      segment%tr_Reg%Tr(n)%is_initialized=.true.
+    enddo
+  endif
+
+  do n = 1, num_fields
+    if ((segment%field(n)%name == 'U' .or. segment%field(n)%name == 'Uamp') .and. &
+         modulo(turns, 2) /= 0) then
+      segment%field(n)%buffer_dst(:,:,:) = -segment%field(n)%buffer_dst(:,:,:)
+      if (segment%is_E_or_W) then
+        segment%normal_trans(:,:,:) = -segment%normal_trans(:,:,:)
+        segment%normal_vel(:,:,:) = -segment%normal_vel(:,:,:)
+        segment%normal_vel_bt(:,:) = -segment%normal_vel_bt(:,:)
+        if (allocated(segment%nudged_normal_vel)) &
+            segment%nudged_normal_vel(:,:,:) = -segment%nudged_normal_vel(:,:,:)
+      else
+        if (allocated(segment%tangential_vel)) &
+            segment%tangential_vel(:,:,:) = -segment%tangential_vel(:,:,:)
+        if (allocated(segment%nudged_tangential_vel)) &
+            segment%nudged_tangential_vel(:,:,:) = -segment%nudged_tangential_vel(:,:,:)
+      endif
+    elseif ((segment%field(n)%name == 'V' .or. segment%field(n)%name == 'Vamp') .and. &
+         modulo(turns, 4) == 3) then
```

Comment:
There is an asymmetry between lines 6223 and 2238 that I do not understand.  For a u-field, the sign of `buffer_dst` and the other fields is flipped for rotations or 1 or 3 turns, but for a v-field it is only flipped if there are 3 turns.  Logically, it seems like there should be the same number of cases where the signs are flipped for both u and v fields.

Reply:
I agree that this asymmetry has now been corrected.